huaweicloud/services/cfw/resource_huaweicloud_cfw_ip_address_group_member.go

```diff
@@ -0,0 +1,246 @@
+	}
+	d.SetId(id.(string))
+
+	return resourceIPAddressGroupMemberRead(ctx, d, meta)
+}
+
+func buildCreateIPAddressGroupMemberBodyParams(d *schema.ResourceData) map[string]interface{} {
+	item := map[string]interface{}{
+		"name":         utils.ValueIngoreEmpty(d.Get("name")),
+		"address":      utils.ValueIngoreEmpty(d.Get("address")),
+		"address_type": utils.ValueIngoreEmpty(d.Get("address_type")),
+		"description":  utils.ValueIngoreEmpty(d.Get("description")),
+	}
+
+	bodyParams := map[string]interface{}{
+		"set_id":        utils.ValueIngoreEmpty(d.Get("group_id")),
+		"address_items": []map[string]interface{}{item},
+	}
+	return bodyParams
+}
+
+func resourceIPAddressGroupMemberRead(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+	region := cfg.GetRegion(d)
+
+	var mErr *multierror.Error
+
+	// getIPAddressGroupMember: Query the CFW IP address group member detail
+	var (
+		getIPAddressGroupMemberHttpUrl = "v1/{project_id}/address-items"
+		getIPAddressGroupMemberProduct = "cfw"
+	)
+	getIPAddressGroupMemberClient, err := cfg.NewServiceClient(getIPAddressGroupMemberProduct, region)
+	if err != nil {
+		return diag.Errorf("error creating CFW Client: %s", err)
+	}
+
+	getIPAddressGroupMemberPath := getIPAddressGroupMemberClient.Endpoint + getIPAddressGroupMemberHttpUrl
+	getIPAddressGroupMemberPath = strings.ReplaceAll(getIPAddressGroupMemberPath, "{project_id}",
+		getIPAddressGroupMemberClient.ProjectID)
+
+	getIPAddressGroupMemberqueryParams := buildGetIPAddressGroupMemberQueryParams(d)
+	getIPAddressGroupMemberPath += getIPAddressGroupMemberqueryParams
+
+	getIPAddressGroupMemberOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+		OkCodes: []int{
+			200,
+		},
+	}
+	getIPAddressGroupMemberResp, err := getIPAddressGroupMemberClient.Request("GET", getIPAddressGroupMemberPath,
+		&getIPAddressGroupMemberOpt)
+
+	if err != nil {
+		return common.CheckDeletedDiag(d, err, "error retrieving IPAddressGroupMember")
+	}
+
+	getIPAddressGroupMemberRespBody, err := utils.FlattenResponse(getIPAddressGroupMemberResp)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	d.SetId(utils.PathSearch("data.records[0].item_id", getIPAddressGroupMemberRespBody, nil).(string))
+
+	mErr = multierror.Append(
+		mErr,
+		d.Set("region", region),
+		d.Set("group_id", utils.PathSearch("data.set_id", getIPAddressGroupMemberRespBody, nil)),
+		d.Set("name", utils.PathSearch("data.records[0].name", getIPAddressGroupMemberRespBody, nil)),
+		d.Set("address", utils.PathSearch("data.records[0].address", getIPAddressGroupMemberRespBody, nil)),
+		d.Set("address_type", utils.PathSearch("data.records[0].address_type", getIPAddressGroupMemberRespBody, nil)),
+		d.Set("description", utils.PathSearch("data.records[0].description", getIPAddressGroupMemberRespBody, nil)),
+	)
+
+	return diag.FromErr(mErr.ErrorOrNil())
+}
+
+func buildGetIPAddressGroupMemberQueryParams(d *schema.ResourceData) string {
```

Comment:
Shall we expose this function and reference it in the acc test?

Reply:
the parameter type in different, can't use this func in acc test